person.rb

```diff
@@ -20,4 +27,42 @@ def of_age?
   def can_use_services?
     is_of_age? || @parent_permission
   end
+
+  def correct_name
+    @name
+  end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
```

Comment:
- [ ] [OPTIONAL] I think it would make your code more modularized if you extract the classes into separate files